content/blog/2016/2016-04-07-pipeline-for-runs-on-hardware.adoc

```diff
@@ -0,0 +1,258 @@
+---
+layout: post
+title: Automating test runs on hardware with Pipeline-as-Code. Part 1
+tags:
+- jenkins
+- jenkins2
+- pipeline
+- embedded
+author: oleg_nenashev
+---
+
+In addition to Jenkins development, during last 8 years I've been involved into continuous integration for hardware and embedded projects.
+At JUC2015/London 
+link:https://www.cloudbees.com/jenkins/juc-2015/presentations/JUC-2015-Europe-Jenkins-Based-CI-for-Nenashev.pdf[I have conducted a talk] about common automation challenges in the area.
+
+In this blog post I would like to concentrate on link:https://jenkins.io/doc/pipeline/[Pipeline] (fka Workflow), which is a new automation-as-code ecosystem for Jenkins.
+It will be proposed by default in the incoming link:/2.0/[Jenkins 2.0 release].
+
+First time I've tried Pipeline two and half years ago, and unfortunately it didn't work for my cases at all. 
+I was very disappointed and tried it only one year later, At this point the plugin became much more stable and useful. 
+It also attracted much more contributors, and it started evolving rapidly and finally became a huge ecosystem powered by Jenkins.
+
+Currently Pipeline is one of the best automation-as-code solutions available in automation servers, so I would like to highlight several Pipeline features, which may be interesting to Jenkins engineers working in the area.
```

Comment:
> Currently Pipeline is one of the best automation-as-code solutions available in automation servers, so I would like to highlight several Pipeline features, which may be interesting to Jenkins engineers working in the area.

Please drop "automation-as-code", that's not a phrase we're using anywhere else (very intentionally). If you're going to make a statement that it's the "best" solution across all "automation servers" I would expect some evidence or comparisons to be made to support that claim. Since that's not what we're talking about on the Jenkins blog, I think that sentence should be dropped. Also "in the area", what area? I think a noun is missing here, presumably "Embedded projects"

Perhaps:

> Currently, Pipeline a powerful tool available for Jenkins users to implement a variety of software delivery pipelines in code. I would like to highlight several Pipeline features which may be interesting to Jenkins users working specifically with embedded projects.